src/lib/reasoners/matching.ml

```diff
@@ -482,19 +482,10 @@ module Make (X : Arg) : S with type theory = X.t = struct
     Debug.match_pats_modulo pat lsubsts;
     List.fold_left (match_one_pat mconf env tbox pat) [] lsubsts
 
-  let pat_weight s t =
-    let sf = (Expr.term_view s).f in
-    let tf = (Expr.term_view t).f in
-    match sf, tf with
-    | Symbols.Name _, Symbols.Op _ -> -1
-    | Symbols.Op _, Symbols.Name _ -> 1
-    | _ -> Expr.(depth t - depth s)
-
   let matching mconf env tbox pat_info =
     let open Matching_types in
     let pats = pat_info.trigger in
     let pats_list = pats.E.content in
-    Options.heavy_assert (fun () -> Lists.is_sorted pat_weight pats_list);
```

Comment:
Why did you remove this assert? In the previous code, we have to sort triggers in different locations, so the design was a bit fragile but now we have to ensure that `Expr` always produce sorted pattern list. We can replace this assertion by documentation.

Reply:
Because `pat_weight` is now private to the `Expr` module and I am not sure it is worth exposing just for this assert. I did forget to add documentation to `trigger.content`, will do.